### PR TITLE
Don't set CSIMigrationAWS in k8s 1.27

### PR DESCRIPTION
**Pull request by rifelpet**
All AWS e2e jobs using CI builds of k/k started failing after the AWS in-tree provider was removed:

https://testgrid.k8s.io/kops-ipv6#kops-aws-cni-cilium-ipv6

https://storage.googleapis.com/kubernetes-jenkins/logs/e2e-kops-aws-cni-cilium-ipv6/1633186496192188416/artifacts/2600:1f13:831:2b01:7a9c:e176:b0fb:800d/kubelet.log

`E0307 19:29:58.709805    3441 run.go:74] "command failed" err="failed to set feature gates from initial flags-based config: unrecognized feature gate: CSIMigrationAWS"`

This removes the feature gate starting in k8s 1.27

### CODE REVIEW

**Comment by hakman** [comment]
Do we still need `InTreePluginAWSUnregister`?

**Comment by rifelpet** [comment]
> Do we still need `InTreePluginAWSUnregister`?

The PR only removed CSIMigrationAWS: https://github.com/kubernetes/kubernetes/pull/115838/files#diff-71e3b98f9a6bbf5b8421e26a7ba0c079f397cd8d49abacdad943c66a4f44f03d I'd like to try to get e2e passing and then we can do further cleanup if needed.

**Review by hakman — APPROVED** [review]

File: tests/e2e/pkg/tester/skip_regex.go [hunk 1]
```diff
@@ -62,8 +62,6 @@ func (t *Tester) setSkipRegexFlag() error {
 		skipRegex += "|TCP.CLOSE_WAIT"
 		// https://github.com/cilium/cilium/issues/15361
 		skipRegex += "|external.IP.is.not.assigned.to.a.node"
-		// https://github.com/cilium/cilium/issues/14287
-		skipRegex += "|same.port.number.but.different.protocols|same.hostPort.but.different.hostIP.and.protocol"
```
Comment (hakman) [inline tests/e2e/pkg/tester/skip_regex.go:66]: Guess this is why GCE tests are failing:
```
Summarizing 2 Failures:
  [FAIL] [sig-network] HostPort [It] validates that there is no conflict between pods with same hostPort but different hostIP and protocol [LinuxOnly] [Conformance]
  [FAIL] [sig-network] Services [It] should be able to update service type to NodePort listening on same port number but different protocols

**Review by hakman — CHANGES_REQUESTED** [review]
/lgtm cancel

**Review by olemarkus — APPROVED** [review]

**Comment by hakman** [comment]
😁
/lgtm cancel

File: tests/e2e/pkg/tester/skip_regex.go (hunk 1 above)
Comment (rifelpet) [reply tests/e2e/pkg/tester/skip_regex.go]: oops, had that sitting in my local repo for cilium testing

**Pull request merged**